src/AutoRest.CSharp.V3/Output/Builders/ClientBuilder.cs

```diff
@@ -311,6 +328,33 @@ private Paging GetClientMethodPaging(Method method, Method nextPageMethod, IDict
             }
         }
 
+        private static LongRunningOperation BuildLongRunningOperation(Method method, Response originalResponse, IDictionary<object, object> options)
+        {
+            var originalResponseParameter = new Parameter(
+                "originalResponse",
+                "The original response from starting the operation.",
+                new CSharpType(typeof(AzureResponse)),
+                null,
+                true);
+            var httpMessageParameter = new Parameter(
+                "createOriginalHttpMessage",
+                "Creates the HTTP message used for the original request.",
+                new CSharpType(typeof(Func<>), new CSharpType(typeof(HttpMessage))),
+                null,
+                true);
+            FinalStateVia finalStateVia = GetFinalStateVia(options.GetValue<string>("final-state-via"));
+            string name = $"{method.Name}Operation";
+            return new LongRunningOperation(method, originalResponse, name, new[] { originalResponseParameter, httpMessageParameter }, finalStateVia);
+        }
+
+        private static FinalStateVia GetFinalStateVia(string? rawValue) => rawValue switch
+        {
+            "azure-async-operation" => FinalStateVia.AzureAsyncOperation,
+            "location" => FinalStateVia.Location,
+            "original-uri" => FinalStateVia.OriginalUri,
+            _ => FinalStateVia.Location
```

Comment:
Any reason we are not throwing for unknown values?

Reply:
Location is the default. It used to be azure-async-operation, but was changed which is why some tests need further investigation by Laurent. This value (final-state-via) is optional when declaring an LRO.

Reply:
Let's distinguish between no value and unknown value here and throw for unknown.

Reply:
Done.